core/src/main/scala/kafka/utils/ToolsUtils.scala

```diff
@@ -56,8 +56,14 @@ object ToolsUtils {
     }
     println(s"\n%-${maxLengthOfDisplayName}s   %s".format("Metric Name", "Value"))
     sortedMap.foreach {
-      case (metricName, value) =>
-        println(s"%-${maxLengthOfDisplayName}s : %.3f".format(metricName, value))
+      case (metricName, value) => {
+        val fmt =
+          if (value.isInstanceOf[java.lang.Float] || value.isInstanceOf[java.lang.Double])
```

Comment:
Should we apply scala match pattern?
```scala
      case (metricName, value) =>
        val fmt = value match {
          case _ @ (_: java.lang.Float | _: java.lang.Double) => s"%-${maxLengthOfDisplayName}s : %.3f"
          case _ => s"%-${maxLengthOfDisplayName}s : %s"
        }
        println(fmt.format(metricName, value))
```

Reply:
We can also avoid some duplication. The only thing that differs is the format specifier, the rest should be the same and can be after the `if/else`.

Reply:
@tombentley Is comment from @ijuma addressed?

Reply:
@chia7712 I used the code you pasted above, which factors out the `fmt`.

Reply:
@tombentley please take a look at following code.

```scala
        val specifier = value match {
          case _ @ (_: java.lang.Float | _: java.lang.Double) => "%.3f"
          case _ => "%s"
        }
        println(s"%-${maxLengthOfDisplayName}s : $specifier".format(metricName, value))
```

@ijuma please correct me if I misunderstand your comment :(

Reply:
@chia7712 @ijuma now fixed, thanks!